pkg/iac/terraform/state/enumerator/file_test.go

```diff
@@ -20,18 +20,58 @@ func TestFileEnumerator_Enumerate(t *testing.T) {
 				Path: "testdata/states",
 			},
 			want: []string{
+				"testdata/states/lambda/lambda.tfstate",
+				"testdata/states/s3/terraform.tfstate",
+				"testdata/states/terraform.tfstate",
+			},
+		},
+		{
+			name: "subfolder nesting glob",
+			config: config.SupplierConfig{
+				Path: "testdata/states/**/*.tfstate",
+			},
+			want: []string{
+				"testdata/states/lambda/lambda.tfstate",
+				"testdata/states/s3/terraform.tfstate",
+				"testdata/states/terraform.tfstate",
+			},
+		},
+		{
+			name: "ext glob ?",
+			config: config.SupplierConfig{
+				Path: "testdata/states/?(s3|lambda)/*.tfstate",
+			},
+			want: []string{
+				"testdata/states/lambda/lambda.tfstate",
+				"testdata/states/s3/terraform.tfstate",
+			},
+		},
+		{
+			name: "ext glob @",
+			config: config.SupplierConfig{
+				Path: "testdata/states/**/*.@(tfstate|state)",
+			},
+			want: []string{
+				"testdata/states/lambda/lambda.tfstate",
 				"testdata/states/route53/directory/route53.state",
 				"testdata/states/s3/terraform.tfstate",
 				"testdata/states/terraform.tfstate",
 			},
 		},
+		{
+			name: "ext glob !",
+			config: config.SupplierConfig{
+				Path: "testdata/states/**/*.!(backup)",
+			},
+			err: "Unsupported syntax !()",
+		},
 		{
 			name: "symlinked folder",
 			config: config.SupplierConfig{
 				Path: "testdata/symlink",
 			},
 			want: []string{
-				"testdata/states/route53/directory/route53.state",
+				"testdata/states/lambda/lambda.tfstate",
 				"testdata/states/s3/terraform.tfstate",
 				"testdata/states/terraform.tfstate",
 			},
```

Comment:
Shouldn't the expected value contain `testdata/symlink/` directory ?

Reply:
no we don't follow symlink I didn't change this behavior

Reply:
Understood!